kiota_http/kiota_client_factory.py

```diff
@@ -1,52 +1,63 @@
-import functools
-from sqlite3 import connect
+from __future__ import annotations
+
+from typing import List, Optional
 
 import httpx
 
 from .middleware import (
     AsyncKiotaTransport,
+    BaseMiddleware,
     MiddlewarePipeline,
     ParametersNameDecodingHandler,
     RedirectHandler,
     RetryHandler,
 )
 
+DEFAULT_CONNECTION_TIMEOUT: int = 30
+DEFAULT_REQUEST_TIMEOUT: int = 100
+
 
 class KiotaClientFactory:
-    DEFAULT_CONNECTION_TIMEOUT: int = 30
-    DEFAULT_REQUEST_TIMEOUT: int = 100
 
-    def create_with_default_middleware(self) -> httpx.AsyncClient:
+    @staticmethod
+    def create_with_default_middleware() -> httpx.AsyncClient:
```

Comment:
break this method in two, the additional method would be get_default_client, which returns the client with the default options but no middleware. (public)
And this one would call in the new method.

Reply:
Also check https://github.com/microsoft/kiota-http-python/pull/21/commits/33d2209ae32b88021839b585770ec403aba8e1c3